seawatch_registration/views.py

```diff
@@ -78,3 +80,24 @@ def show_profile(request):
   profile = request.user.profile
 
   return render(request, 'show-profile.html', {'profile': profile})
+
+
+def profileposition_form(request):
+    json = {}
+    try:
+        Profile.objects.get(user=request.user)
+        form = ProfilePositionForm(user=request.user)
+        if request.method == 'POST':
+            form = ProfilePositionForm(request.POST, user=request.user)
+            if form.is_valid():
+                profile = form.cleaned_data['profile']
+                requested_positions = form.cleaned_data['requested_positions']
+                for position in requested_positions:
+                    profileposition = ProfilePosition(profile=profile, position=Position.objects.get(name=position), requested=True, approved=False)
+                    profileposition.save()
+                return render(request,
+                              'position.html',
+                              {'form': ProfilePositionForm(user=request.user), 'success': True})
+        return render(request, 'position.html', {'form': form})
+    except ObjectDoesNotExist:
+        return redirect('/accounts/login/')
```

Comment:
Hier könnten diese Class-Based-Views helfen den Code schlanker zu machen. da gibt es für jede HTTP-Methode direkt eine Methode:
```
from django.http import HttpResponse
from django.views import View

class MyView(View):

    def get(self, request, *args, **kwargs):
        return HttpResponse('Hello, World!')
```
https://docs.djangoproject.com/en/2.2/ref/class-based-views/base/

Eine weitere Möglichkeit wäre ein View-Decorator:
```
from django.views.decorators.http import require_http_methods

@require_http_methods(["POST"])
def profileposition_form(request):
```

Hier würde der Benutzer ja auch weitergeleitet wenn er eingeloggt ist, aber noch kein Profil hat.
Es gibt auch einen `@login_required` Decorator
https://docs.djangoproject.com/en/2.2/topics/auth/default/#the-login-required-decorator